### PR TITLE
Add API approval tests

### LINKED ISSUE #141: Add API approval tests
See serilog or destructurama repos as examples. 

@ArieGato I can post PR if ok.

### PR DESCRIPTION
**Pull request by sungam3r**
fixes #141

### CODE REVIEW

File: tests/Serilog.Sinks.RabbitMQ.Tests/Approval/Serilog.Sinks.RabbitMQ.approved.txt [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                    System.Net.Security.SslPolicyErrors sslAcceptablePolicyErrors = 0,
+                    bool sslCheckCertificateRevocation = false,
+                    int batchPostingLimit = 0,
+                    System.TimeSpan period = default,
+                    Serilog.Formatting.ITextFormatter formatter = null,
+                    bool autoCreateExchange = false,
+                    int maxChannels = 64) { }
+    }
+}
+namespace Serilog.Sinks.RabbitMQ
```
Comment (sungam3r) [inline tests/Serilog.Sinks.RabbitMQ.Tests/Approval/Serilog.Sinks.RabbitMQ.approved.txt:55]: @ArieGato As you can see there are a lot of public APIs here that should not be exposed.

**Comment by sungam3r** [comment]
For now I see 
изображение

If I understand correctly pipeline has not even started because of review settings. So I have no change to fix errors if any - I just don't see them. So I should wait for review. If you are concerned about security, then it is not necessary to require a review. Nobody can just merge change into the repository. Only those who have write access can - owners, collaborators, etc.

**Comment by sungam3r** [comment]
@ArieGato I resolved conflicts.

**Review by ArieGato — CHANGES_REQUESTED** [review]

File: tests/Serilog.Sinks.RabbitMQ.Tests/Serilog.Sinks.RabbitMQ.Tests.csproj [hunk 2]
```diff
@@ -12,6 +12,8 @@
     <PackageReference Include="FluentAssertions" Version="6.12.0" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.8.0" />
     <PackageReference Include="NSubstitute" Version="5.1.0" />
+    <PackageReference Include="PublicApiGenerator" Version="11.1.0" />
+    <PackageReference Include="Shouldly" Version="4.2.1" />
```
[the remark this answers is not in the document]
Comment (sungam3r) [reply tests/Serilog.Sinks.RabbitMQ.Tests/Serilog.Sinks.RabbitMQ.Tests.csproj:16]: They are not for the same thing. Shouldly has `ShouldMatchApproved` method specifically for comparing files and I use this method in `PublicApi_Should_Not_Change_Unintentionally` test. You can see the same approach in the main project https://github.com/serilog/serilog/blob/dev/test/Serilog.ApprovalTests/ApiApprovalTests.cs#L19.

[the remark this answers is not in the document]
Comment (ArieGato) [reply tests/Serilog.Sinks.RabbitMQ.Tests/Serilog.Sinks.RabbitMQ.Tests.csproj:16]: What I mean is that they are both frameworks for asserting test results.  I really don't want to mix them in other unit tests. There still are some Assert from the Microsoft test library. 

If this functionality is not supported in FluentAssertions then we can evaluate whether we can convert towards Shouldly.

**Review by ArieGato — APPROVED** [review]

**Comment by ArieGato** [comment]
@sungam3r  There are some conflicts

**Comment by sungam3r** [comment]
Done.

File: tests/Serilog.Sinks.RabbitMQ.Tests/Serilog.Sinks.RabbitMQ.Tests.csproj (hunk 2 above)
[the remark this answers is not in the document]
Comment (sungam3r) [reply tests/Serilog.Sinks.RabbitMQ.Tests/Serilog.Sinks.RabbitMQ.Tests.csproj]: #150

**Pull request merged**